### PR TITLE
Add profiling to coalton-library/system

### PR DESCRIPTION
**Pull request by Izaakwltn**
This adds a profiling interface (besides just Time) for Coalton.

### CODE REVIEW

**Review by jbouwman — APPROVED** [review]
looks good to me

File: library/system.lisp [hunk 1]
```diff
@@ -8,6 +8,12 @@
    #:time
    #:sleep)
   (:export
+   #:get-real-time
+   #:get-run-time
+   #+sbcl #:get-bytes-consed
```
Comment (stylewarning) [inline library/system.lisp:13]: this should be exported by all implementations but maybe return None

File: library/system.lisp [hunk 2]
```diff
@@ -62,6 +68,64 @@ While the result will always contain microseconds, some implementations may retu
...
+    (lisp UFix ()
+      (cl:get-internal-run-time)))
+
+  (declare get-real-time (Unit -> UFix))
+  (define (get-real-time)
+    "Gets the real-time."
+    (lisp UFix ()
+      (cl:get-internal-real-time)))
+
+  #+sbcl
```
Comment (stylewarning) [inline library/system.lisp:89]: please do not do this in the standard library

**Review by stylewarning — COMMENTED** [review]
The conditionals were a mistake.

File: library/system.lisp (hunk 2 above)
Comment (Izaakwltn) [reply library/system.lisp:89]: won't happen again

**Pull request merged**